pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -211,7 +230,7 @@ func (r *ReconcileClusterClaim) Reconcile(ctx context.Context, request reconcile
 	if lifetime != nil {
 		logger.WithField("lifetime", lifetime).Debug("checking whether lifetime of ClusterClaim has elapsed")
 		pendingCond := controllerutils.FindClusterClaimCondition(claim.Status.Conditions, hivev1.ClusterClaimPendingCondition)
-		if pendingCond != nil && pendingCond.Status == corev1.ConditionFalse {
+		if pendingCond.Status == corev1.ConditionFalse {
```

Comment:
Will this break for existing claims with no pending condition? If yes, we can keep `pendingCond != nil` as extra safety net

Reply:
This will never be nil because as a part of reconcile we are always checking if those conditions are not present then [initialize](https://github.com/openshift/hive/pull/1408/files/3c57723f538887a124852bffd9b7dd0d9070c852..48b81ce9c350521fed4a2872f215284d90af44bd#diff-429aa8363e2e77526949e5194551e15dc6b97dd27ca7f9dbdd9ed9b963f8609eR179-R190) them.

Reply:
ah, yes